packages/consul/src/Consul.js

```diff
@@ -158,9 +158,11 @@ export class Consul {
   }
 
   async register({ data, options = {} } = {}) {
-    if (!data.name || !data.id) throw new Error('must provide name and id to register for consul service discovery');
+    if (options.onBeforeRegister) {
```

Comment:
what's that onBeforeRegister/onAfterRegister? is it being used?
do we need it?

Reply:
Yes, I'm going to add additional logs through this.

Reply:
im not sure we want log on every interval of checking register of consul

Reply:
If we had it, I would save at least few days which I've spent to understand what's happened. 
I'm strongly convinced that we should log it...